napari/layers/utils/layer_utils.py

```diff
@@ -807,7 +807,9 @@ def resize(
         if size < current_size:
             self.remove(range(size, current_size))
         elif size > current_size:
-            to_append = pd.concat([self._defaults] * (size - current_size))
+            to_append = self._defaults.reindex(
+                index=range(size - current_size), method='pad'
+            )
             self.append(to_append)
```

Comment:
really nice! [docs for anyone interested](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.reindex.html)